test/operations/candidates/upload_file_test.rb

```diff
@@ -0,0 +1,240 @@
+# frozen_string_literal: true
+
+require "test_helper"
+
+class Candidates::UploadFileTest < ActionDispatch::IntegrationTest
+  setup do
+    ActsAsTenant.current_tenant = tenants(:toughbyte_tenant)
+
+    @empty_pdf_file = ActionDispatch::Http::UploadedFile.new(
+      filename: "empty.pdf",
+      type: "application/pdf",
+      tempfile: Rails.root.join("test/fixtures/files/empty.pdf")
+    )
+
+    @not_empty_pdf_file = ActionDispatch::Http::UploadedFile.new(
+      filename: "cv_with_links.pdf",
+      type: "application/pdf",
+      tempfile: Rails.root.join("test/fixtures/files/cv_with_links.pdf")
+    )
+  end
+
+  test "should set source and text_checksum metadata when we upload a pdf file" do
+    candidate = candidates(:sam)
+
+    assert_equal candidate.files.size, 0
+
+    Candidates::UploadFile.new(
+      candidate:,
+      file: @empty_pdf_file,
+      source: "linkedin",
+      namespace: :api
+    ).call.value!
+
+    candidate.reload
+
+    assert_equal candidate.files.size, 1
+
+    file = candidate.files.first
```

Comment:
```suggestion
    file = candidate.files.take
```

Reply:
i'd prefer first because there are tests where we take second

Reply:
Yes, but don't you assert that candidate has exactly 1 file?

Reply:
i do.
i see what you mean, but still would like to use `first`